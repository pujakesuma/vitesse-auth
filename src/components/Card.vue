<script setup lang="ts">
interface Props {
  customClass?: string
}

// eslint-disable-next-line vue/no-setup-props-destructure
const { customClass = 'w-full lg:w-3/5' } = defineProps<Props>()

</script>

<template>
  <div class="rounded-lg lg:rounded-l-lg lg:rounded-r-none shadow-2xl opacity-75" :class="customClass">
    <div class="p-4 md:p-12 text-center lg:text-left">
      <slot />
    </div>
  </div>
</template>
