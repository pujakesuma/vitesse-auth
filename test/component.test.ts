import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Card from '../src/components/Card.vue'

describe('Card.vue', () => {
  it('should render', () => {
    const wrapper = mount(Card, { props: { customClass: 'w-full lg:w-3/5' } })
    expect(wrapper.text()).toContain('')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async() => {
    const wrapper = mount(Card, { props: { customClass: '' } })
    expect(wrapper.text()).toContain('')
    expect(wrapper.text()).toContain('')
  })
})
