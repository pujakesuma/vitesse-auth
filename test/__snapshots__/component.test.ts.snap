// Vitest Snapshot v1

exports[`Card.vue > should render 1`] = `
"<div class=\\"rounded-lg lg:rounded-l-lg lg:rounded-r-none shadow-2xl opacity-75 w-full lg:w-3/5\\">
  <div class=\\"p-4 md:p-12 text-center lg:text-left\\"></div>
</div>"
`;